controller/work_item_labels_blackbox_test.go

```diff
@@ -0,0 +1,227 @@
+}
+
+func (l *TestWorkItemLabelREST) TearDownTest() {
+	l.clean()
+}
+
+func (l *TestWorkItemLabelREST) SecuredController() (*goa.Service, app.WorkitemController) {
+	pub, _ := wittoken.RSAPublicKey()
+	svc := testsupport.ServiceAsUser("WorkItemLabel-Service", wittoken.NewManager(pub), l.testIdentity)
+	return svc, NewWorkitemController(svc, l.db, l.Configuration)
+}
+
+func (l *TestWorkItemLabelREST) UnSecuredController() (*goa.Service, app.WorkitemController) {
+	svc := goa.New("WorkItemLabel-Service")
+	return svc, NewWorkitemController(svc, l.db, l.Configuration)
+}
+
+func (l *TestWorkItemLabelREST) TestAttachDetachLabelToWI() {
+	wiCnt := 2
+	lblCnt := 3
+	fixtures := tf.NewTestFixture(l.T(), l.DB, tf.Spaces(1), tf.Iterations(1), tf.Areas(1), tf.WorkItems(wiCnt), tf.Labels(lblCnt))
+	svc, ctrl := l.SecuredController()
+	relatedLink := fmt.Sprintf("/%s/labels", fixtures.WorkItems[0].ID)
+
+	// Fetch WI and verify Labels Relationship
+	_, fetchedWI := test.ShowWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, nil, nil)
+	require.NotNil(l.T(), fetchedWI.Data.Relationships.Labels)
+	require.NotNil(l.T(), fetchedWI.Data.Relationships.Labels.Links)
+	assert.Contains(l.T(), *fetchedWI.Data.Relationships.Labels.Links.Related, relatedLink)
+	assert.Empty(l.T(), fetchedWI.Data.Relationships.Labels.Data)
+
+	u := app.UpdateWorkitemPayload{
+		Data: &app.WorkItem{
+			ID:   &fixtures.WorkItems[0].ID,
+			Type: APIStringTypeWorkItem,
+			Attributes: map[string]interface{}{
+				"version": fixtures.WorkItems[0].Version,
+			},
+			Relationships: &app.WorkItemRelationships{},
+		},
+	}
+	// attach 2 labels
+	apiLabelType := label.APIStringTypeLabels
+	lbl0 := fixtures.Labels[0].ID.String()
+	lbl1 := fixtures.Labels[1].ID.String()
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{
+			{
+				ID:   &lbl0,
+				Type: &apiLabelType,
+			},
+			{
+				ID:   &lbl1,
+				Type: &apiLabelType,
+			},
+		},
+	}
+	_, updatedWI := test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
+	require.NotNil(l.T(), updatedWI.Data.Relationships.Labels.Links)
+	assert.Contains(l.T(), *updatedWI.Data.Relationships.Labels.Links.Related, relatedLink)
+	assert.Len(l.T(), updatedWI.Data.Relationships.Labels.Data, 2)
+	mustHave := map[string]struct{}{
+		lbl0: {},
+		lbl1: {},
+	}
+	for _, lblData := range updatedWI.Data.Relationships.Labels.Data {
+		delete(mustHave, *lblData.ID)
+	}
+	require.Empty(l.T(), mustHave)
+
+	// attach 1 label
+	lbl2 := fixtures.Labels[2].ID.String()
+	u.Data.Attributes["version"] = updatedWI.Data.Attributes["version"]
+	u.Data.Relationships.Labels = &app.RelationGenericList{
+		Data: []*app.GenericData{
+			{
+				ID:   &lbl2,
+				Type: &apiLabelType,
+			},
+		},
+	}
+	_, updatedWI = test.UpdateWorkitemOK(l.T(), svc.Context, svc, ctrl, fixtures.WorkItems[0].ID, &u)
+	assert.NotNil(l.T(), updatedWI)
```

Comment:
should be `require.NoNil()`

Reply:
done in https://github.com/fabric8-services/fabric8-wit/pull/1627/commits/d669b65ccc5f6d78c3bc11494e6065ee6341559c